### PR TITLE
Extend test suite for interops

### LINKED ISSUE #1578: Add tests for interops that don't have any tests at the moment
`Neo.Crypto.VerifyWithECDsaSecp256k1`, `Neo.Crypto.CheckMultisigWithECDsaSecp256k1`, `System.Enumerator.Concat`, `System.Enumerator.Create`, `System.Iterator.*`, `System.Binary.Deserialize`, `System.Blockchain.GetTransactionHeight`, `System.Runtime.GetCallingScriptHash`, `System.Runtime.GetEntryScriptHash`, `System.Runtime.Platform`, `System.Runtime.Log`, `System.Runtime.GetTime`, `System.Storage.Delete`, `System.Storage.AsReadOnly`, `System.Contract.Destroy`.

### PR DESCRIPTION
**Pull request by fyrchik**
Close #1578 .

### CODE REVIEW

File: pkg/core/interop_system_test.go [hunk 1]
```diff
@@ -826,6 +925,22 @@ func TestContractUpdate(t *testing.T) {
...
+	v.LoadScriptWithHash(cs.Script, cs.Hash, smartcontract.All)
+	require.NoError(t, contractDestroy(ic)) // silent error when contract is missing
+	require.NoError(t, ic.DAO.PutContractState(cs))
+
+	v.Estack().PushVal("key")
+	v.Estack().PushVal("value")
+	require.NoError(t, storageGetContext(ic))
+	require.NoError(t, storagePut(ic))
+	require.NoError(t, contractDestroy(ic))
+	require.Error(t, storageGetContext(ic))
```
Comment (roman-khimov) [inline pkg/core/interop_system_test.go:941]:
```
storageDelete()
contractDestroy(cs.Hash)
```
?

File: pkg/crypto/keys/private_key.go [hunk 2]
```diff
@@ -20,16 +20,21 @@ type PrivateKey struct {
...
 
-// NewPrivateKey creates a new random Secp256k1 private key.
+// NewPrivateKey creates a new random Secp256r1 private key.
 func NewPrivateKey() (*PrivateKey, error) {
-	priv, x, y, err := elliptic.GenerateKey(elliptic.P256(), rand.Reader)
+	return NewPrivateKeyOnCurve(elliptic.P256())
+}
+
+// NewPrivateKeyOnCurve creates a new random private key using curve c.
+func NewPrivateKeyOnCurve(c elliptic.Curve) (*PrivateKey, error) {
```
Comment (roman-khimov) [inline pkg/crypto/keys/private_key.go:29]: I'm not sure we can safely allow any `elliptic.Curve` here. We really only work with r1 and k1, so maybe it's enough to make this internal and expose `NewK1PrivateKey()` along with the old `NewPrivateKey` to the outside world.

File: pkg/core/interop/crypto/hash_test.go [hunk 3]
```diff
@@ -1,46 +1,69 @@
...
+	t.Run("bad message", func(t *testing.T) {
+		ic := &interop.Context{VM: vm.New()}
+		ic.VM.Estack().PushVal(stackitem.NewArray(nil))
+		require.Error(t, interopFunc(ic))
+	})
+}
+
 func TestSHA256(t *testing.T) {
-	// 0x0100 hashes to 47dc540c94ceb704a23875c11273e16bb0b8a87aed84de911f2133568115f254
-	res := "47dc540c94ceb704a23875c11273e16bb0b8a87aed84de911f2133568115f254"
```
Comment (roman-khimov) [inline pkg/core/interop/crypto/hash_test.go:18]: There is a huge difference between this old test and current one and I'd rather still have static known in-out pairs for hashes.

File: pkg/crypto/keys/private_key.go (hunk 2 above)
Comment (fyrchik) [reply pkg/crypto/keys/private_key.go:29]: Fixed.

File: pkg/core/interop/crypto/hash_test.go (hunk 3 above)
Comment (fyrchik) [reply pkg/core/interop/crypto/hash_test.go:18]: Fixed.

File: pkg/core/interop_system_test.go (hunk 1 above)
Comment (fyrchik) [reply pkg/core/interop_system_test.go:943]: This was about checking that items were also deleted during `contractDestroy`.
I have added checks for that using DAO.

**Review by roman-khimov — APPROVED** [review]

**Pull request merged**